lambdas/call-ticf-cri/src/main/java/uk/gov/di/ipv/core/callticfcri/CallTicfCriHandler.java

```diff
@@ -2,17 +2,173 @@
+        this.userIdentityService = new UserIdentityService(configService);
+        this.ticfCriService = new TicfCriService(configService);
+        this.ciMitService = new CiMitService(configService);
+        this.criStoringService =
+                new CriStoringService(
+                        configService,
+                        new AuditService(AuditService.getDefaultSqsClient(), configService),
+                        null,
+                        new VerifiableCredentialService(configService),
+                        ciMitService);
+    }
+
+    public CallTicfCriHandler(
+            ConfigService configService,
+            IpvSessionService ipvSessionService,
+            ClientOAuthSessionDetailsService clientOAuthSessionDetailsService,
+            UserIdentityService userIdentityService,
+            TicfCriService ticfCriService,
+            CiMitService ciMitService,
+            CriStoringService criStoringService) {
+        this.configService = configService;
+        this.ipvSessionService = ipvSessionService;
+        this.clientOAuthSessionDetailsService = clientOAuthSessionDetailsService;
+        this.userIdentityService = userIdentityService;
+        this.ticfCriService = ticfCriService;
+        this.ciMitService = ciMitService;
+        this.criStoringService = criStoringService;
+    }
+
     @Override
     @Tracing
     @Logging(clearState = true)
-    public Map<String, Object> handleRequest(JourneyRequest input, Context context) {
-        return null;
+    public Map<String, Object> handleRequest(ProcessRequest input, Context context) {
+        LogHelper.attachComponentIdToLogs(configService);
+        LogHelper.attachCriIdToLogs(TICF_CRI);
+
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(input);
+            String featureSet = RequestHelper.getFeatureSet(input);
+            String journeyType = RequestHelper.getJourneyType(input);
+            configService.setFeatureSet(featureSet);
+            IpvSessionItem ipvSessionItem = ipvSessionService.getIpvSession(ipvSessionId);
+            ClientOAuthSessionItem clientOAuthSessionItem =
+                    clientOAuthSessionDetailsService.getClientOAuthSession(
+                            ipvSessionItem.getClientOAuthSessionId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSessionItem.getGovukSigninJourneyId());
+
+            List<SignedJWT> ticfVcs =
+                    ticfCriService.getTicfVc(
+                            clientOAuthSessionItem,
+                            ipvSessionItem,
+                            journeyType.equals(REUSE_JOURNEY_TYPE)
+                                    ? List.of()
+                                    : userIdentityService.getUserIssuedCredentials(
+                                            clientOAuthSessionItem.getUserId()));
```

Comment:
Interesting - so the plan is that we'll attach a 'journey type' to the state in the journey map :+1:

It's worth noting that this logic may have to change in future - it _should_ be 'send all VCs collected in this session', but that does boil down to 'is it reuse or not'.

I think that's fine for now, and we can update later - we might have to add journey ID to the VC store item to allow us to do so (which may be useful for other reasons as well).

Reply:
Interesting. I went this way to avoid having to look at the users current state to determine if they're on a reuse journey or not. But the journey ID thing is interesting. Not for now though, as you say.